src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,234 @@
+
+  if (noise->Dim() > 0) {
+    AddVectorsOfUnequalLength(*noise, signal);
+  }
+}
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+
+    const char *usage =
+        "Corrupts the wave files supplied via input pipe with the specified\n"
+        "room-impulse response (rir_matrix) and additive noise distortions\n"
+        "(specified by corresponding files).\n";
+
+    ParseOptions po(usage);
+    std::string noise_file;
+    BaseFloat snr_db = 20;
+    bool multi_channel_output = false;
+    int input_channel = 0;
+    int rir_channel = 0;
+    int noise_channel = 0;
+    bool normalize_output = true;
+    BaseFloat volume = 0;
+
+    po.Register("multi-channel-output", &multi_channel_output,
+                "Specifies if the output should be multi-channel or not");
+    po.Register("input-wave-channel", &input_channel,
+                "Specifies the channel to be used from input as only a "
+                "single channel will be used to generate reverberated output");
+    po.Register("rir-channel", &rir_channel,
+                "Specifies the channel of the room impulse response, "
+                "it will only be used when multi-channel-output is false");
+    po.Register("noise-channel", &noise_channel,
+                "Specifies the channel of the noise file, "
+                "it will only be used when multi-channel-output is false");
+    po.Register("noise-file", &noise_file,
+                "File with additive noise");
+    po.Register("snr-db", &snr_db,
+                "Desired SNR(dB) of the output");
+    po.Register("normalize-output", &normalize_output,
+                "If true, then after reverberating and "
+                "possibly adding noise, scale so that the signal "
+                "energy is the same as the original input signal.");
+    po.Register("volume", &volume,
+                "If nonzero, a scaling factor on the signal that is applied "
+                "after reverberating and possibly adding noise. "
+                "If you set this option to a nonzero value, "
+                "it will be as if you had also specified --normalize-output=false.");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() != 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    if (multi_channel_output) {
+      if (rir_channel != 0 || noise_channel != 0)
+        KALDI_ERR << "Do not set --rir-channel or --noise-channel "
+                  << "when --multi-channel-output is true.";
+    }
+
+    std::string input_wave_file = po.GetArg(1);
+    std::string rir_file = po.GetArg(2);
+    std::string output_wave_file = po.GetArg(3);
+
+    WaveData input_wave;
+    {
+      Input ki(input_wave_file);
+      input_wave.Read(ki.Stream());
+    }
+
+    const Matrix<BaseFloat> &input_matrix = input_wave.Data();
+    BaseFloat samp_freq_input = input_wave.SampFreq();  // read sampling fequency
+    int32 num_samp_input = input_matrix.NumCols(),  // number of samples in recording
```

Comment:
use consistent terminology. you are using "recording", "input_wave" to refer to the same thing